library/unwind/build.rs

```diff
@@ -2,8 +2,13 @@ use std::env;
 
 fn main() {
     println!("cargo:rerun-if-changed=build.rs");
-    let target = env::var("TARGET").expect("TARGET was not set");
 
+    if env::var_os("CARGO_CFG_MIRI").is_some() {
```

Comment:
Should there be a rerun-if-changed for this, or does cargo rerun anyway when the cfg flags change?

Reply:
A dedicated rerun-if-change definitely won't hurt, I think.

That said, presumably you actually *don't* want to re-run the script (and rebuild the world) when switching between --cfg=miri and not, if you've previously had it build successfully? RUST_CHECK has similar behavior where once we've built LLVM, it never gets passed down (even in check builds).

I admit I've not followed the story around cfg(miri) enough to know where we're setting that or how, can you point me to that?

Reply:
> A dedicated rerun-if-change definitely won't hurt, I think.

Done.

> I admit I've not followed the story around cfg(miri) enough to know where we're setting that or how, can you point me to that?

Miri sets `--cfg miri` on all things it builds.
Miri also uses a separate target dir, so for a given target dir, it should never be the case that it is sometimes built with `--cfg miri` and sometimes without.

> RUST_CHECK has similar behavior where once we've built LLVM, it never gets passed down (even in check builds).

So that is some further extra magic and rustbuild implements?

Reply:
Ah, okay, if this never differs for a given target dir then it should be OK -- no need for special magic.

Yes, RUST_CHECK is only set by rustbuild (not Cargo or others); definition [here](https://github.com/rust-lang/rust/blob/master/src/bootstrap/builder.rs#L1384-L1394).